src/selvage/test/flow.clj

```diff
@@ -0,0 +1,199 @@
+(ns selvage.test.flow
+  "clojure.test version of the flow macro"
+   (:require [schema.core :as s]
+             [selvage.core :as core]
+             [clojure.spec.test.alpha :as spec.test]
+             [clojure.test :as t]
+             [selvage.visibility :as vis]
+             [taoensso.timbre :as timbre])
+  (:import [java.io ByteArrayOutputStream PrintStream StringWriter]))
+
+(def ^:dynamic *probe-timeout* 300)
+(def ^:dynamic *probe-sleep-period* 10)
+(def ^:dynamic *verbose* false)
+(def ^:dynamic *world* {})
+(def ^:dynamic *flow* {})
+
+(defn worlds [] (deref core/worlds-atom))
+
+(def ^:dynamic *intermediate-report*)
+
+(defn diff-test-result
+  "Subtract two clojure.test style summary maps."
+  [before after]
+  {:pass  (apply - (map :pass [after before]))
+   :error (apply - (map :error [after before]))
+   :test  (apply - (map :test [after before]))
+   :fail  (apply - (map :fail [after before]))})
+
+(defn report-count []
+  (diff-test-result *intermediate-report* @t/*report-counters*))
+
+(defn update-report-counters [new-reports]
+  (when t/*report-counters*
+    (dosync (commute t/*report-counters*
+                     (fn [current-counters] (merge-with + current-counters new-reports))))))
+
+(defmacro with-report-counters
+  [& body]
+  `(update-report-counters
+     (binding [*intermediate-report* (or (some-> t/*report-counters* deref) ~t/*initial-report-counters*)]
+       (binding [t/*report-counters* (ref *intermediate-report*)]
+         ~@body
+         (diff-test-result *intermediate-report* @t/*report-counters*)))))
```

Comment:
I'm not sure how well this works with parallelism, but let's get to that when it comes up, because I couldn't think of another way to do this